From 5c8d2b00e5b80e8a2ef0f68f44e0617c462432fc Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:44 -0400
Subject: [tools] perf/timechart: Use proc_num to implement --power-only

Message-id: <1408953210-26343-37-git-send-email-jolsa@redhat.com>
Patchwork-id: 88745
O-Subject: [PATCH RHEL7.1 BZ1133083 036/322] perf timechart: Use proc_num to implement --power-only
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 753c505dc49a87a4421d452bda048e4b93e8e42b
Author: Stanislav Fomichev <stfomichev@yandex-team.ru>
Date: Fri Nov 1 20:25:47 2013 +0400

description
===========
Don't use special flag to indicate power-only mode, just set proc_num to
0.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-timechart.c b/tools/perf/builtin-timechart.c
index c352be4..6410c9e 100644
--- a/tools/perf/builtin-timechart.c
+++ b/tools/perf/builtin-timechart.c
@@ -912,7 +912,7 @@ static int determine_display_tasks(u64 threshold)
   /* no exit marker, task kept running to the end */
   if (p->end_time == 0)
    p->end_time = last_time;
-  if (p->total_time >= threshold && !power_only)
+  if (p->total_time >= threshold)
    p->display = 1;
 
   c = p->all;
@@ -923,7 +923,7 @@ static int determine_display_tasks(u64 threshold)
    if (c->start_time == 1)
     c->start_time = first_time;
 
-   if (c->total_time >= threshold && !power_only) {
+   if (c->total_time >= threshold) {
     c->display = 1;
     count++;
    }
@@ -950,6 +950,8 @@ static void write_svg_file(const char *filename)
 
  numcpus++;
 
+ if (power_only)
+  proc_num = 0;
 
  /* We'd like to show at least proc_num tasks;
   * be less picky if we have fewer */
@@ -967,9 +969,11 @@ static void write_svg_file(const char *filename)
   svg_cpu_box(i, max_freq, turbo_frequency);
 
  draw_cpu_usage();
- draw_process_bars();
+ if (proc_num)
+  draw_process_bars();
  draw_c_p_states();
- draw_wakeups();
+ if (proc_num)
+  draw_wakeups();
 
  svg_close();
 }
-- 
1.7.1